### PR TITLE
Add raw public key doc.

### PR DESCRIPTION
**Pull request by jyao1**
Fix https://github.com/DMTF/libspdm/issues/2233

### CODE REVIEW

File: doc/raw_public_key.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Raw Public Key
+
+## Specification
+
+libspdm raw public key follows [RFC 7250](https://www.rfc-editor.org/rfc/rfc7250) defined public key format.
+
+## Crypto Library
+
+libspdm calls crypto library directly to parse the raw public key.
+- For openssl, libspdm calls `d2i_RSA_PUBKEY_bio` for RSA, `d2i_EC_PUBKEY_bio` for ECDSA, and `d2i_PUBKEY_bio` for EdDSA and SM2DSA.
```
Comment (steven-bellock) [inline doc/raw_public_key.md:10]: openssl -> OpenSSL
mbedtls -> Mbed TLS

File: doc/raw_public_key.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+## Generation
+
+openssl can be used to generate the DER based raw public key.
+```
+openssl pkey -in end_point.key.priv.pem -inform PEM -pubout -outform DER -out end_point.key.pub.der
+```
+
+## Registration
+
+libspdm can use `LIBSPDM_DATA_PEER_PUBLIC_KEY` and `LIBSPDM_DATA_LOCAL_PUBLIC_KEY` to register the raw public key based authentication.
```
Comment (steven-bellock) [inline doc/raw_public_key.md:22]: libspdm -> The Integrator

File: doc/raw_public_key.md (hunk 1 above)
Comment (jyao1) [reply doc/raw_public_key.md]: done.

File: doc/raw_public_key.md (hunk 2 above)
Comment (jyao1) [reply doc/raw_public_key.md]: Good catch. Fixed.

**Pull request merged**